src/contexts/Payouts/index.tsx

```diff
@@ -165,49 +167,123 @@ export const PayoutsProvider = ({
+        oldClaimedRewards[ledger.stash] = erasToCheck
           .map((e) => e.toString())
           .filter((r: string) => validatorExposedEras(ledger.stash).includes(r))
-          .filter((r: string) => !erasClaimed.includes(r));
+          .filter((r: string) => erasClaimed.includes(r));
+      }
+    }
+
+    // Reformat old unclaimed rewards to match new format.
+    const oldClaimedByEra: Record<string, string[]> = {};
+    erasToCheck.forEach((era) => {
+      const eraValidators: string[] = [];
+      Object.entries(oldClaimedRewards).forEach(([validator, eras]) => {
+        if (eras.includes(era)) {
+          eraValidators.push(validator);
+        }
+      });
+      if (eraValidators.length > 0) {
+        oldClaimedByEra[era] = eraValidators;
+      }
+    });
+
+    const unclaimedRewards: Record<string, string[]> = {};
+
+    // Accumulate calls to fetch unclaimed rewards for each era for all validators.
+    const unclaimedRewardsEntries = erasToCheck
+      .map((era) => uniqueValidators.map((v) => [era, v]))
+      .flat();
+
+    const results = await Promise.all(
+      unclaimedRewardsEntries.map(([era, v]) =>
+        api.query.staking.claimedRewards<AnyApi>(era, v)
+      )
+    );
+
+    for (let i = 0; i < results.length; i++) {
+      const pages = results[i].toHuman() || [];
+      const era = unclaimedRewardsEntries[i][0];
+      const validator = unclaimedRewardsEntries[i][1];
+      const exposure = getLocalEraExposure(network.name, era, activeAccount);
+      const exposedPage =
+        exposure?.[validator]?.exposedPage !== undefined
+          ? String(exposure[validator].exposedPage)
+          : '0';
+
+      // Add to `unclaimedRewards` if payout page has not yet been claimed.
+      if (!pages.includes(exposedPage)) {
+        if (unclaimedRewards?.[validator]) {
+          unclaimedRewards[validator].push(era);
+        } else {
+          unclaimedRewards[validator] = [era];
+        }
       }
     }
 
     // Reformat unclaimed rewards to be { era: validators[] }.
-    const unclaimedByEra: Record<string, string[]> = {};
+    let unclaimedByEra: Record<string, string[]> = {};
     erasToCheck.forEach((era) => {
       const eraValidators: string[] = [];
       Object.entries(unclaimedRewards).forEach(([validator, eras]) => {
-        if (eras.includes(era)) eraValidators.push(validator);
+        if (eras.includes(era)) {
+          eraValidators.push(validator);
+        }
       });
-      if (eraValidators.length > 0) unclaimedByEra[era] = eraValidators;
+      if (eraValidators.length > 0) {
+        unclaimedByEra[era] = eraValidators;
+      }
     });
 
+    function mergeClaimedData(
+      unclaimedByEra: Record<string, string[]>,
+      claimedByEra: Record<string, string[]>
+    ): Record<string, string[]> {
+      const result = { ...unclaimedByEra };
+
+      for (const era in claimedByEra) {
+        if (claimedByEra.hasOwnProperty(era) && result.hasOwnProperty(era)) {
```

Comment:
Iterating over own properties
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in#iterating_over_own_properties_only
I'd use `in` operator here - `era in result`